apps/sync/test/unit/adapters/history_test.js

```diff
@@ -0,0 +1,329 @@
+  MockNavigatorDatastore,
+  setup,
+  sinon,
+  suite,
+  suiteSetup,
+  suiteTeardown,
+  teardown,
+  test
+*/
+
+require('/shared/test/unit/mocks/mock_lazy_loader.js');
+require('/shared/test/unit/mocks/mock_navigator_datastore.js');
+require('/apps/system/test/unit/mock_asyncStorage.js');
+requireApp('sync/js/adapters/history.js');
+
+window.DataAdapters = {};
+
+suite('sync/adapters/history >', () => {
+  var realDatastore, realLazyLoader, realAsyncStorage, testCollectionData;
+  var updatePlacesSpy;
+  var kintoCollection = {
+    list() {
+      return Promise.resolve({
+        data: testCollectionData
+      });
+    }
+  };
+
+  function getPlacesStore() {
+    return navigator.getDataStores().then(stores => {
+      return Promise.resolve(stores[0]);
+    });
+  }
+
+  function verifyPlaces(collectionItem, placesItem) {
+    var payload = collectionItem.payload;
+    assert.equal(payload.histUri, placesItem.url);
+    assert.equal(payload.title, placesItem.title);
+    for (var i = 0; i < payload.visits.length; i++) {
+      assert.equal(payload.visits[i].date, placesItem.visits[i] * 1000);
+    }
+    assert.equal(payload.id, placesItem.fxsyncId);
+  }
+
+  function testDataGenerator(initIndex, initDate, count) {
+    var list = [];
+    for (var i = initIndex; i < initIndex + count; i++) {
+      var visits = [];
+      var startData = initDate + i * 100;
+      for (var j = 0; j < 3; j++) {
+        visits.push({
+          date: (startData + j * 10) * 1000, type: 3
+        });
+      }
+
+      list.unshift({
+        id: 'UNIQUE_ID_' + i,
+        last_modified: initDate + i * 10,
+        payload: {
+          id: 'UNIQUE_ID_' + i,
+          histUri: 'http://example' + i + '.com/',
+          title: 'Example ' + i + ' Title',
+          visits: visits
+        }
+      });
+    }
+    return list;
+  }
+
+  suiteSetup(() => {
+    realDatastore = navigator.getDataStores;
+
+    realAsyncStorage = window.asyncStorage;
+    window.asyncStorage = MockasyncStorage;
+
+    realLazyLoader = window.LazyLoader;
+    window.LazyLoader = MockLazyLoader;
+  });
+
+  suiteTeardown(() => {
+    navigator.getDataStores = realDatastore;
+
+    window.LazyLoader = realLazyLoader;
+
+    window.asyncStorage = realAsyncStorage;
+  });
+
+  setup(() => {
+    navigator.getDataStores = MockNavigatorDatastore.getDataStores;
+    updatePlacesSpy = sinon.spy(HistoryHelper, 'updatePlaces');
+    testCollectionData = [];
+  });
+
+  teardown(() => {
+    updatePlacesSpy.restore();
+    MockDatastore._inError = false;
+    MockDatastore._records = Object.create(null);
+    window.asyncStorage.mTeardown();
+  });
+
+  test('update - empty records', done => {
+    var historyAdapter = DataAdapters.history;
+    historyAdapter.update(kintoCollection).then((result) => {
+      assert.equal(result, false);
+      assert.equal(updatePlacesSpy.callCount, 0);
+      assert.equal(asyncStorage.mItems[HISTORY_COLLECTION_MTIME], null);
+    }).then(done, done);
```

Comment:
In case of rejection, we need to fail the test.
    .then(done, reason => {
        assert.ok(false, reason);
    });

Do this here and in the rest of the tests, please.